File: tests/fixtures/cpp/method_calls.cpp
#include <string>

class Processor {
public:
    Processor(const std::string& data) : data_(data) {}

    std::string process() const {
        return data_;
    }

    std::string validate(const std::string& input) const {
        return input;
    }

private:
    std::string data_;
};

void run() {
    Processor p("test");
    std::string result = p.process();
    std::string valid = p.validate(result);
}

File: tests/fixtures/javascript/method_calls.js
class Processor {
    constructor(data) {
        this.data = data;
    }

    process() {
        return this.data;
    }

    validate(input) {
        return input;
    }
}

function run() {
    const p = new Processor("test");
    const result = p.process();
    const valid = p.validate(result);
    return valid;
}
